subgit/core.py

```diff
@@ -486,6 +487,111 @@ def pull(self, names):
+
+            if answer:
+                self.repo_deletion(active_repos=active_repos)
+
+        if isinstance(repo_names, list):  # If at least one name was specified
+
+            repo_choices = ", ".join(repo_names)
+
+            answer = self.yes_no(f"Are you sure you want to delete the following repos '{repo_choices}'")
+
+            if answer:
+                self.repo_deletion(active_repos, repo_names)
+
+    def repo_deletion(self, active_repos, repos=None):
+        """
+        Helper method that recieves a list of repos to delete and takes care of the logic
+        """
+        if not repos:
+            repos = active_repos
+
+        in_cnf_file = True
+
+        dirty_repos = False
+
+        repo_paths = []
+
+        path_no_exist = []
+
+        good_repos = []
+
+        for name in repos:
+            repo_paths.append(os.path.join(os.getcwd(), name))
+            if name not in active_repos:
+                log.warning(f"Argument: '{name}' does not exist in '.subgit.yml'")
+                in_cnf_file = False
+
+        if not in_cnf_file:
+            return 1
+
+        for path in repo_paths:
+
+            if not os.path.exists(path):
+                log.warning(f'Path to repo does not exist: {os.path.basename(path)} | Skipping {os.path.basename(path)}')
+                path_no_exist.append(path)
+                repo_paths.remove(path)
+
+        for path in repo_paths:
+
+            if not self.is_valid_repo(path):
+                log.warning(f"FATAL: '{os.path.basename(path)}' is not a git repo")
+                return 1
+
+        for path in repo_paths:
+
+            current_repo = Repo(path)
+
+            if self.check_remote(current_repo, os.path.basename(path)):
+                dirty_repos = True
+            else:
+                good_repos.append(path)
+
+        if not dirty_repos:
+            for repo in good_repos:
+                if path not in path_no_exist:
+                    shutil.rmtree(repo)
+                    log.info(f'Successfully removed repo: {os.path.basename(repo)}')
+
+    def check_remote(self, repo, base_name):
+        """
+        Takes repo object and name of directory to delete
+        """
+        is_diff = False
+        # has_new = repo.remotes.origin.refs["master"].commit != repo.heads[branch].commit
+        for remote in repo.remotes:
+            for branch in repo.branches:
+                if remote.refs[str(branch)].commit != repo.heads[str(branch)].commit or repo.is_dirty(untracked_files=True):
+                    is_diff = True
+                    log.warning(f"FATAL: '{base_name}' has some diff(s) in the local repo or the remote that needs be taken care of before deletion.")
+
+        return is_diff
+
+    def is_valid_repo(self, path):
+        try:
+            if Repo(path):
+                return True
+        except git.exc.InvalidGitRepositoryError:
+            return False
+
```

Comment:
So technically here  you could end up with a case where your if case would not pass but also not throw an exception. In that case it would default out to `None` as return value.

```suggestion
    def is_valid_repo(self, path):
        try:
            Repo(path)
        except git.exc.InvalidGitRepositoryError:
            return False
        
        return True

```